src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs

```diff
@@ -42,83 +42,78 @@ public override void Initialize(AnalysisContext context)
-
-            protected DisposableFieldAnalyzer(Compilation compilation)
+            INamedTypeSymbol namedType = (INamedTypeSymbol)ctx.Symbol;
+            if (disposeAnalysisHelper.IsDisposable(namedType))
             {
-                DisposeAnalysisHelper.TryGetOrCreate(compilation, out _disposeAnalysisHelper!);
-                RoslynDebug.Assert(_disposeAnalysisHelper != null);
+                return;
             }
 
-            public void AnalyzeSymbol(SymbolAnalysisContext symbolContext)
+            var disposableFields = namedType
+                .GetMembers()
+                .OfType<IFieldSymbol>()
+                .Where(f => !f.IsStatic && disposeAnalysisHelper.IsDisposable(f.Type))
+                .ToSet();
+            if (disposableFields.Count == 0)
             {
-                INamedTypeSymbol namedType = (INamedTypeSymbol)symbolContext.Symbol;
-                if (_disposeAnalysisHelper.IsDisposable(namedType))
-                {
-                    return;
-                }
+                return;
+            }
 
-                IEnumerable<IFieldSymbol> disposableFields = from member in namedType.GetMembers()
-                                                             where member.Kind == SymbolKind.Field && !member.IsStatic
-                                                             let field = member as IFieldSymbol
-                                                             where _disposeAnalysisHelper.IsDisposable(field.Type)
-                                                             select field;
-                if (!disposableFields.Any())
-                {
-                    return;
-                }
+            var disposableFieldNamesBuilder = ImmutableArray.CreateBuilder<string>();
 
-                var disposableFieldsHashSet = new HashSet<ISymbol>(disposableFields);
-                IEnumerable<TTypeDeclarationSyntax> classDecls = GetClassDeclarationNodes(namedType, symbolContext.CancellationToken);
-                foreach (TTypeDeclarationSyntax classDecl in classDecls)
+            ctx.RegisterOperationAction(context => AnalyzeOperation(context, namedType, disposableFields, disposableFieldNamesBuilder), OperationKind.SimpleAssignment, OperationKind.FieldInitializer);
+            ctx.RegisterSymbolEndAction(context => AnalyzeSymbolEnd(context, disposableFieldNamesBuilder.ToImmutable()));
+        }
+
+        private static void AnalyzeOperation(OperationAnalysisContext ctx, INamedTypeSymbol parent, ISet<IFieldSymbol> disposableFields, ImmutableArray<string>.Builder disposableFieldNamesBuilder)
+        {
+            if (ctx.Operation is IAssignmentOperation { Target: IFieldReferenceOperation field } assignment
+                && IsObjectCreation(assignment.Value)
+                && !ctx.Options.IsConfiguredToSkipAnalysis(Rule, field.Field.Type, parent, ctx.Compilation)
+                && disposableFields.Contains(field.Field))
```

Comment:
I would reverse the order of these 2 checks. `IsConfiguredToSkipAnalysis` is a comparatively expensive operation, so should be delayed as much as possible.